api/src/main/java/com/epam/pipeline/manager/cluster/autoscale/ScaleDownHandler.java

```diff
@@ -198,6 +212,32 @@ private void log(final PipelineRun run, final String text) {
                 .build());
     }
 
+    private boolean isNodeAliveOnCloud(final Node node) {
+        final String instanceId = getNodeCloudInstanceId(node);
+        final String label = getNodeLabel(node);
+        final Optional<InstanceTerminationState> state;
+        final boolean existsOnCloud;
+        if (isPooled(node)) {
+            final NodeRegionLabels cloudRegion = kubernetesManager.getNodeRegion(label);
+            state = cloudFacade.getInstanceTerminationState(cloudRegion, instanceId);
+            existsOnCloud = cloudFacade.instanceExists(cloudRegion, instanceId);
+        } else {
+            final Long runId = Long.parseLong(label);
+            final Long cloudRegionId = pipelineRunManager.findRun(runId)
+                    .map(PipelineRun::getInstance)
+                    .map(RunInstance::getCloudRegionId)
+                    .orElseThrow(
+                            () -> new IllegalArgumentException(
+                                    String.format("Can't find region by runId: %d", runId))
+                    );
+            state = cloudFacade.getInstanceTerminationState(cloudRegionId, instanceId);
+            existsOnCloud = cloudFacade.instanceExists(cloudRegionId, instanceId);
+        }
```

Comment:
Do we really need to do two requests here? Because both instance existence check and termination state retrieval is e.g. `describe instances` request on AWS.

It seems like we can just use a single call to `CloudFacade::getInstanceState` here.

Reply:
My idea here is that if we can't find `terminationState` for the instance, there could be two cases for that:
 - Instance save and sound
 - Instance already gone from the console
As I understand, both cases are relevant here. So, to be sure that instance is really terminated we also need to check that it isn't exists on a cloud.